kiwi/system/root_bind.py

```diff
@@ -98,6 +98,28 @@ def mount_kernel_file_systems(self):
                 '%s: %s' % (type(e).__name__, format(e))
             )
 
+    def umount_kernel_file_systems(self):
+        """
+        Umount kernel filesystems
+
+        :raises KiwiMountKernelFileSystemsError: if some kernel filesystem
+            fails to mount
+        """
+        umounts = [
+            mnt for mnt in self.mount_stack if mnt.device
+            in self.bind_locations
+        ]
+        for umount in reversed(umounts):
+            if umount.is_mounted():
+                try:
+                    umount.umount_lazy()
+                    self.mount_stack.remove(umount)
+                except Exception as e:
+                    log.warning(
+                        'Image root directory %s not cleanly umounted: %s',
+                        self.root_dir, format(e)
+                    )
+
```

Comment:
This duplicates some code that it already present in the private _cleanup_mount_stack() method. Maybe you could consider to extend _cleanup_mount_stack such that you can pass mountpoints to umount. This would then result in a smaller umount_kernel_file_systems that might look like this:

```python
def umount_kernel_file_systems(self):
    self._cleanup_mount_stack(self.bind_locations)
```

Reply:
Good point, gonna fix it.

Reply:
This is great now :+1: